test/flows/find_coronavirus_support_flow_test.rb

```diff
@@ -0,0 +1,597 @@
+      end
+
+      should "render extra guidance for Scotland" do
+        add_responses nation: "scotland"
+
+        assert_rendered_outcome text: "Get help from Women’s Aid Scotland"
+      end
+
+      should "render extra guidance for Wales" do
+        add_responses nation: "wales"
+
+        assert_rendered_outcome text: "Get help from Live Fear Free"
+      end
+    end
+
+    context "paying_bills" do
+      setup do
+        add_responses need_help_with: %w[paying_bills],
+                      afford_rent_mortgage_bills: "yes",
+                      nation: "england"
+      end
+
+      should "render paying_bills if need_help_with? is paying_bills" do
+        assert_rendered_outcome text: "If you’re finding it hard to pay your rent, mortgage or bills"
+      end
+
+      should "render extra guidance for England" do
+        assert_rendered_outcome text: "(Shelter)"
+      end
+
+      should "render extra guidance for Scotland" do
+        add_responses nation: "scotland"
+
+        assert_rendered_outcome text: "(Shelter Scotland)"
+      end
+
+      should "render extra guidance for Wales" do
+        add_responses nation: "wales"
+
+        assert_rendered_outcome text: "(Shelter Cymru)"
+      end
+
+      should "render extra guidance for Northern Ireland" do
+        add_responses nation: "northern_ireland"
+
+        assert_rendered_outcome text: "Get help from Citizens Advice"
+      end
+    end
+
+    context "getting_food" do
+      setup do
+        add_responses need_help_with: %w[getting_food],
+                      afford_food: "yes",
+                      get_food: "yes",
+                      nation: "england"
+      end
+
+      should "render afford_food if afford_food is yes" do
+        assert_rendered_outcome text: "If you’re finding it hard to afford food"
+      end
+
+      should "render extra afford_food guidance for England" do
+        assert_rendered_outcome text: "Find out if you can get help from a foodbank"
+      end
+
+      should "render extra afford_food guidance for Scotland" do
+        add_responses nation: "scotland"
+
+        assert_rendered_outcome text: "Scottish Welfare Fund"
+      end
+
+      should "render extra afford_food guidance for Wales" do
+        add_responses nation: "wales"
+
+        assert_rendered_outcome text: "(Welsh Government)"
+      end
+
+      should "render extra afford_food guidance for Northern Ireland" do
+        add_responses nation: "northern_ireland"
+
+        assert_rendered_outcome text: "Community Helpline for Northern Ireland"
+      end
+
+      should "render get_food if get_food is no" do
+        add_responses get_food: "no"
+
+        assert_rendered_outcome text: "If you’re unable to get food or medicine"
+      end
+    end
+
+    context "being_unemployed" do
+      setup do
+        add_responses need_help_with: %w[being_unemployed],
+                      self_employed: "no",
```

Comment:
This won't work. If self_employed is "yes", the have_you_been_made_unemployed question is skipped:

next_node do
        if calculator.self_employed == "yes"
          question calculator.next_question(:have_you_been_made_unemployed)
        else
          question :have_you_been_made_unemployed
        end
      end


So to test rendering of the have_you_been_made_unemployed results, self_employed needs to be "no".

I did it in this order, i.e. have_you_been_made_unemployed first as have_you_been_made_unemployed is the first calculator value checked in the being_unemployed partial.

Also, I thought it was ok to only test the nation specific content once.

Reply:
I suppose I could put the tests in separate sub contexts? One where `self_employed` is "yes" and the other where it's "no" to test the `have_you_been_made_unemployed` outcome. I think that might make it easier to read, though I'm not sure I relish the idea of having two sets of nation checks.

Reply:
Oh I see, thanks for explaining. I hadn't realised that it was actually a one or the other for unemployed or self employed rather than that both can be available (shame the code wasn't an `else`). I guess with hindsight this should probably be two different partials.

> Also, I thought it was ok to only test the nation specific content once.

Yeah, that seemed a reasonable approach if testing one meant they were all executed. We've got that slightly annoying situation here where, by only testing them once for this partial, that either the unemployed conditionals are executed or the self employed ones are. So we lose some coverage either way.

I can understand not wanting to do all of the nation conditionals again and this seems sufficiently nuanced that it's not something that could be enforced into future or explained. So don't worry about them if we can't express a way that they add value.

Reply:
So, do you think it would be better to split the `being_unemployed` tests into separate contexts, or are they ok as they are?

Reply:
If it's not too much of a pain I think it'd be better to split the being_unemployed into a separate context. I think it would be good to satisfy the convention of full coverage through these files.